packages/client/src/app/stores/auth.js

```diff
@@ -0,0 +1,58 @@
+import { defineStore } from 'pinia'
+import { router } from '../app-router'
+import axios from '../axios'
+
+export const useAuthStore = defineStore({
+  id: 'auth',
+  state: () => ({
+    user: null,
+    loading: true
+  }),
+  getters: {
+    isAuthenticated: (state) => state.user !== null
+  },
+  actions: {
+    async register(data) {
+      await axios
+        .post('/user', data)
+        .then(() => {
+          router.push({ name: 'login' })
+        })
+        .catch((error) => {
+          console.error(error)
+        })
+    },
+    async login(data) {
+      await axios
+        .post('/user/auth', data)
+        .then((res) => {
+          this.user = res.data.user
+          router.push({ name: 'home' })
+        })
+        .catch((error) => {
+          console.error(error)
+        })
+    },
+    async logout() {
+      this.user = null
+
+      await axios
+        .post('/user/logout')
+        .then(() => {
+          router.push({ name: 'home' })
+        })
+        .catch((error) => {
+          console.error(error)
+        })
+    },
+    async getLoggedInUser() {
```

Comment:
I am not sure what the purpose of this function will be?

Reply:
this get's the authenticated user and with this we can check if someone is authenticated

Reply:
I renamed it to make it clearer